Library/Formula/json-fortran.rb

```diff
@@ -0,0 +1,73 @@
+class JsonFortran < Formula
+  desc "A Fortran 2008 JSON API"
+  homepage "https://github.com/jacobwilliams/json-fortran"
+  url "https://github.com/jacobwilliams/json-fortran/archive/4.1.1.tar.gz"
+  sha256 "97f258d28536035ef70e9ead5c7053e654106760a12db2cc652587ed61b76124"
+
+  head "https://github.com/jacobwilliams/json-fortran.git"
+
+  option "with-unicode-support", "Build json-fortran to support unicode text in json objects and files"
+  option "with-keg-only", "Keg only install (Skip linking json-fortran.) Required to support multiple Fortran compilers."
+  option "without-make-test", "Skip running build-time tests (not recommended)"
+
+  depends_on "cmake" => :build
+  depends_on "cmake"
+  depends_on "robodoc" => :recommended
+  depends_on :fortran
+
+  keg_only "User selected keg only install" if build.with? "keg-only"
```

Comment:
OK, but my original rational was to allow a user to install json-fortran in such a way that other versions could be installed elsewhere on the system to support other Fortran compilers. I’ll scrap this idea entirely if you like, but additional feedback would be most welcome.

Here is some background on why I attempted to let the user choose a keg-only install:
Fortran module files, and object code are not compatible between compilers (or sometimes even compiler releases from the same vendor). To install json-fortran on a system that wishes to support multiple Fortran compilers, it is necessary to install multiple versions of the library, each one built with the Fortran compiler it supports. To achieve this, the json-fortran project/library defaults to installing as:
```
<DESTDIR | PREFIX>/jsonfortran-${CMAKE_Fortran_COMPILER_ID}-${VERSION}/
├── cmake
│   ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-config-version.cmake
│   ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-config.cmake
│   ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-targets-release.cmake
│   └── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-targets.cmake
└── lib
    ├── json_module.mod
    ├── libjsonfortran.${VERSION}.dylib
    ├── libjsonfortran.${VERSION_MAJOR}.${VERSION_MINOR}.dylib -> libjsonfortran.${VERSION}.dylib
    ├── libjsonfortran.a
    └── libjsonfortran.dylib -> libjsonfortran.${VERSION_MAJOR}.${VERSION_MINOR}.dylib
```
Usually this means that the project is installed in `/usr/local/jsonfortran-${CMAKE_Fortran_COMPILER_ID}-${VERSION}/` and as a consequence homebrew does not create any symbolic links in `/usr/local/lib` and /usr/local/include`. Also, since the `.mod` file is binary, and compiler specific—even in its name—AND the `lib` directory is on most compilers module search paths it is just thrown in with the static and dynamic libraries.

Note, that If the user wishes to only support one Fortran compiler (which is what we anticipate anyone installing via Homebrew will want) and use a more standard GNU install structure they may  set the CMake cache variable `USE_GNU_INSTALL_CONVENTION` to  `TRUE`. This will result in the following install structure:
```
<DESTDIR | PREFIX>/
├── include
│   └── json_module.mod
└── lib
    ├── cmake
    │   └── json-fortran-${VERSION}
    │       ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-config-version.cmake
    │       ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-config.cmake
    │       ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-targets-release.cmake
    │       └── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-targets.cmake
    ├── libjsonfortran.${VERSION}.dylib
    ├── libjsonfortran.${VERSION_MAJOR}.${VERSION_MINOR}.dylib
    ├── libjsonfortran.a
    └── libjsonfortran.dylib
```
The thinking was to provide an option to allow a more “normal” GNU install structure, that is consistent with the other Fortran projects I have seen on Homebrew, and that will allow the libraries and module files to get symlinked by Homebrew. This is the default for the Homebrew installation, but I thought I would give the user the opportunity to use the original install structure which will support multiple compilers. However, if multiple compiler support is desired, the user should probably install json-fortran by hand.